src/server/list_family.cc

```diff
@@ -274,13 +271,11 @@ OpResult<string> OpMoveSingleShard(const OpArgs& op_args, string_view src, strin
     db_slice.PreUpdate(op_args.db_cntx.db_index, dest_it);
   }
 
-  db_slice.PreUpdate(op_args.db_cntx.db_index, src_it);
```

Comment:
check out line 268. It could be that src_it was invalidated by the insertion of dest_it. This is bad. You will call post update on invalid it

Reply:
Very good catch! Fixed.